broadbean/sequence.py

```diff
@@ -0,0 +1,1065 @@
+
+        newdata1 = dict([(key, self.element(key).copy())
+                         for key in self._data.keys()])
+        newdata2 = dict([(key+N, other.element(key).copy())
+                         for key in other._data.keys()])
+        newdata1.update(newdata2)
+
+        newseq._data = newdata1
+
+        newsequencing1 = dict([(key, self._sequencing[key].copy())
+                               for key in self._sequencing.keys()])
+        newsequencing2 = dict()
+
+        for key, item in other._sequencing.items():
+            newitem = item.copy()
+            # update goto and jump according to new sequence length
+            if newitem['goto'] > 0:
+                newitem['goto'] += N
+            if newitem['jump_target'] > 0:
+                newitem['jump_target'] += N
+            newsequencing2.update({key+N: newitem})
+
+        newsequencing1.update(newsequencing2)
+
+        newseq._sequencing = newsequencing1
+
+        newseq._awgspecs = other._awgspecs.copy()
+
+        return newseq
+
+    def copy(self):
+        """
+        Returns a copy of the sequence.
+        """
+        newseq = Sequence()
+        newseq._data = deepcopy(self._data)
+        newseq._meta = deepcopy(self._meta)
+        newseq._awgspecs = deepcopy(self._awgspecs)
+        newseq._sequencing = deepcopy(self._sequencing)
+
+        return newseq
+
+    def setSequenceSettings(self, pos, wait, nreps, jump, goto):
+        """
+        Set the sequence setting for the sequence element at pos.
+
+        Args:
+            pos (int): The sequence element (counting from 1)
+            wait (int): The wait state specifying whether to wait for a
+                trigger. 0: OFF, don't wait, 1: ON, wait. For some backends,
+                additional integers are allowed to specify the trigger input.
+                0 always means off.
+            nreps (int): Number of repetitions. 0 corresponds to infinite
+                repetitions
+            jump (int): Event jump target, the position of a sequence element.
+                If 0, the event jump state is off.
+            goto (int): Goto target, the position of a sequence element.
+                0 means next.
+        """
+
+        warnings.warn('Deprecation warning. This function is only compatible '
+                      'with AWG5014 output and will be removed. '
+                      'Please use the specific setSequencingXXX methods.')
+
+        # Validation (some validation 'postponed' and put in checkConsistency)
+        #
+        # Because of different compliances for different backends,
+        # most validation of these settings is deferred and performed
+        # in the outputForXXX methods
+
+        self._sequencing[pos] = {'twait': wait, 'nrep': nreps,
+                                 'jump_target': jump, 'goto': goto,
+                                 'jump_input': 0}
+
+    def setSequencingTriggerWait(self, pos: int, wait: int) -> None:
```

Comment:
I would rename this to set_trigger_wait. In fact, rename all set functions "setSequencingXXXX" to set_xxxx